src/Profiles.ts

```diff
@@ -336,6 +347,224 @@ export class Profiles {
+            ignoreFocusOut: true,
+            canPickMany: false
+        };
+        // confirm that the user really wants to delete
+        if (await vscode.window.showQuickPick([localize("deleteProfile.showQuickPick.yes", "Yes"),
+            localize("deleteProfile.showQuickPick.no", "No")], quickPickOptions) !== localize("deleteProfile.showQuickPick.yes", "Yes")) {
+            this.log.debug(localize("deleteProfile.showQuickPick.log.debug", "User picked no. Cancelling delete of profile"));
+            return;
+        }
+
+        const profileType = ZoweExplorerApiRegister.getMvsApi(deletedProfile).getProfileTypeName();
+        try {
+            this.deleteProf(deletedProfile, profileName, profileType);
+        } catch (error) {
+            this.log.error(localize("deleteProfile.delete.log.error", "Error encountered when deleting profile! ") + JSON.stringify(error));
+            await errorHandling(error, profileName, error.message);
+            throw error;
+        }
+
+        vscode.window.showInformationMessage("Profile " + profileName + " was deleted.");
+        return profileName;
+    }
+
+    public async deleteProf(ProfileInfo, ProfileName, ProfileType) {
+        let zosmfProfile: IProfile;
+        try {
+            zosmfProfile = await (await this.getCliProfileManager(ProfileType))
+            .delete({ profile: ProfileInfo, name: ProfileName, type: ProfileType });
+        } catch (error) {
+            vscode.window.showErrorMessage(error.message);
+        }
+        return zosmfProfile.profile;
+    }
+
+    public async deleteProfile(
+        datasetTree: IZoweTree<IZoweDatasetTreeNode>, ussTree: IZoweTree<IZoweUSSTreeNode>,
+        jobsProvider: IZoweTree<IZoweJobTreeNode>, node?: IZoweNodeType) {
+
+            let deleteLabel: string;
+            let deletedProfile: IProfileLoaded;
+            if (!node){
+                deletedProfile = await this.getDeleteProfile();
+                if (!deletedProfile) {
+                    return;
+                } else {
+                    deleteLabel = deletedProfile.name;
+                }
+            } else {
+                deletedProfile = node.getProfile();
+                deleteLabel = node.getProfileName();
+            }
+            const deleteSuccess = await this.deletePrompt(deletedProfile);
+            if (!deleteSuccess){
+                vscode.window.showInformationMessage(localize("deleteProfile.noSelected",
+                    "Operation Cancelled"));
+                return;
+            }
+
+            // Delete from Data Set Recall
+            const recallDs: string[] = datasetTree.getRecall();
+            for (let i = recallDs.length - 1; i >= 0; i--) {
+                const findNode = recallDs[i].substring(1, recallDs[i].indexOf("]")).trim();
+                if (findNode === deleteLabel) {
+                    datasetTree.removeRecall(recallDs[i]);
+                }
+            }
```

Comment:
This works. But I think there is a better way
```suggestion
            recallDs.slice().reverse()
                .filter(ds => ds.substring(1, ds.indexOf("]")).trim()  === deleteLabel)
                .forEach(ds => {
                    datasetTree.removeRecall(ds);
            });
```
because _reverse_ mutates the array in place, we call _slice_  to get a new copy of the array before reversing it. `[...recallDs]` would also work in place of the `recallDs.slice()`.